src/txmempool.cpp

```diff
@@ -966,38 +962,34 @@ void CTxMemPool::addUnchecked(const CTxMemPoolEntry &entry, bool validFeeEstimat
 
 void CTxMemPool::UpdateChild(txiter entry, txiter child, bool add)
 {
-    setEntries s;
-    if (add && mapLinks[entry].children.insert(child).second) {
+    CTxMemPoolEntry::Children s;
+    if (add && entry->GetMemPoolChildren().insert(child).second) {
         cachedInnerUsage += memusage::IncrementalDynamicUsage(s);
-    } else if (!add && mapLinks[entry].children.erase(child)) {
+    } else if (!add && entry->GetMemPoolChildren().erase(child)) {
```

Comment:
c977ecc917ecf40c554173a52700973ea7d42f07
Since `add` has `bool` type, could be
```suggestion
    } else if (entry->GetMemPoolChildren().erase(child)) {
```

Reply:
I prefer to not make changes like this unless there's a motivating reason. The mapLinks refactor is about removing mapLinks, code style changes should be proposed separately.

In particular, your recommended change introduces a bug, which is another reason to avoid changes like this.

If the preceding if fails, it is either because add was not set OR because the item was already in the Children set. When we then check the else if, we should only erase if we set add to false, otherwise we will spuriously delete children.

Reply:
My bad ((
Sorry for noise.